### ISSUE TITLE
HDR content performance

### ISSUE THREAD

**Issue by Ptilopsis01** [mpv-player/mpv#13047]
### Important Information

Provide following Information:
- mpv version  v0.37.0-70-g562450f5
- Windows Version  win11 22631.2792
- Source of the mpv binary  https://github.com/zhongfly/mpv-winbuild
- If known which version of mpv introduced the problem  not quite sure
- Possible screenshot or video of visual glitches
![IMG_20231207_043506](https://github.com/mpv-player/mpv/assets/93318731/779a4902-b32e-4060-8404-b32f51ea5b78)
![IMG_20231207_043520](https://github.com/mpv-player/mpv/assets/93318731/66aa6ae0-a8e3-4c3a-a89b-4b259aac2212)

### Reproduction steps

The HDR content behaves low saturation when playing fullscreen(p2), which will not happen with maximum window(p1). But they are the same in the pictures when I tried to take a screen shot, so I have to take photos of the screen. I'm not sure whether it's the problem of my settings or mpv or my monitor.

(I've already turned windows HDR mode on, and my monitor supports HDR400.)

### Expected behavior

HDR content plays correctly when playing fullscreen.

### Log file

[mpv.log](https://github.com/mpv-player/mpv/files/13591500/mpv.log)

### Sample files

Sample files needed to reproduce this issue can be uploaded to https://0x0.st/
or similar sites. (Only needed if the issue cannot be reproduced without it.)
Do not use garbage like "cloud storage", especially not Google Drive.

**Comment by hooke007** [comment]
`format=colorlevels=full` would break all SDR colors, why you set this option.

**Comment by Ptilopsis01** [comment]
> What happens if you use `--no-config --vo=gpu-next`? You have some things in your config that could interfere like `'vf-append' = 'format=colorlevels=full:colormatrix=bt.709:dolbyvision=yes'`

It behaves correctly when using `--no-config --vo=gpu-next`
It also behaves better than the original one with my settings except the `vf-append` attribute (but still different from the one using `--no-config --vo=gpu-next` or maximum window)

**Comment by Ptilopsis01** [comment]
> `format=colorlevels=full` would break all SDR colors, why you set this option. 

So what did this option do? I think that my monitor supports full level, but many videos are "limited", which may behaves strange.

> `'icc-profile-auto' = 'yes'` BTW， I don't think your local icc suport hdr.

Well, actually I set this to "no" in the HDR profile.

**Comment by hooke007** [comment]
It's a fiter to mark the src video to be full range. So mpv would not transfer it correctly when playback.

**Comment by Ptilopsis01** [comment]
By the Way, the window size is larger than the monitor when it's maximum. Maybe it's a bug.

**Comment by hooke007** [comment]
`Setting option 'border' = 'no' (flags = 4)` Because you were using this option.
Not recommended.

**Comment by Ptilopsis01** [comment]
I also found that `target-colorspace-hint` has an effect on the output when playing fullscreen, which caused this:

> It also behaves better than the original one with my settings except the `vf-append` attribute (but still different from the one using `--no-config --vo=gpu-next` or maximum window)

**Comment by kasper93** [comment]
> I also found that `target-colorspace-hint` has an effect on the output when playing fullscreen

It sends HDR data to your display. Which works only in fullscreen.

**Comment by Ptilopsis01** [comment]
> It sends HDR data to your display. Which works only in fullscreen.

So it's not recommended if the monitor is not good at handling HDR contents?

**Comment by hooke007** [comment]
@kasper93 Here is another serious issue about no-border https://github.com/tomasklaen/uosc/issues/811.

**Comment by Jules-A** [comment]
> @kasper93 Here is another serious issue about no-border [tomasklaen/uosc#811](https://github.com/tomasklaen/uosc/issues/811).

Yeah... borderless windowed-fullscreen is super broken in MPV, to keep the taskbar I have to use 
```
snap-window
geometry=2560x1400+0+0
no-border
``` 
which gets it usable but this bug is really annoying...

**Comment by kasper93** [comment]
@hooke007 @Jules-A Works as expected and designed by Microsoft. Use `--no-titlebar --border` if you want to have compositor features.

**Comment by Jules-A** [comment]
> @hooke007 @Jules-A Works as expected and designed by Microsoft. Use `--no-titlebar --border` if you want to have compositor features.

AHAHA no, that's way more broken, if you move or resize the window the titlebar returns, you have an ugly border and part of the window is cut off below the taskbar due to re-addition of the titlebar...

**Comment by kasper93** [comment]
I don't see anything to do here.

See new options and related docs for hints how to configure HDR output best for your display.
https://mpv.io/manual/master/#options-target-colorspace-hint
https://mpv.io/manual/master/#options-target-colorspace-hint-mode
